File: java/doit_java/Chapter8/src/polymorphism/CustomerTest.java
package polymorphism;

public class CustomerTest {
    public static void main(String[] args) {

        Customer customerLim = new Customer();
        customerLim.setCustomerID(10010);
        customerLim.setCustomerName("임채영");
        customerLim.bonusPoint = 1000;
        System.out.println(customerLim.showCustomerInfo());

        Customer customerJung = new VIPCustomer(10020, "정흥규", 1000);
        customerJung.bonusPoint = 1000;
        System.out.println(customerJung.showCustomerInfo());

        System.out.println("\n====== 할인율과 보너스 포인트 계산 ======");

        int price = 10000;
        int limPrice = customerLim.calcPrice(price);
        int jungPrice = customerJung.calcPrice(price);

        System.out.println(customerLim.getCustomerName() + " 님이 " + limPrice + "원 지불 하셨습니다.");
        System.out.println(customerLim.showCustomerInfo());

        System.out.println(customerJung.getCustomerName() + " 님이 " + jungPrice + "원 지불 하셨습니다.");
        System.out.println(customerJung.showCustomerInfo());
    }
}

File: python/p_oop2/chap07/inheritance2/customer_test.py
from customer import Customer
from vip_customer import VIPCustomer


def main():
    customer_lee = Customer(10010, "이순신")
#   customer_lee.set_customer_name("이순신")
#   customer_lee.set_customer_id(10010)
    customer_lee.bonus_point = 1000
    print(customer_lee.show_customer_info())

    customer_kim = VIPCustomer(10020, "김유신")
#   customer_kim.set_customer_name("김유신")
#   customer_kim.set_customer_id(10020)
    customer_kim.bonus_point = 10000
    print(customer_kim.show_customer_info())


if __name__ == '__main__':
    main()
